catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/image/NitfPreStoragePlugin.java

```diff
@@ -223,4 +223,51 @@ private String buildOverviewTitle(String title) {
         outputStream.close();
         return thumbnailBytes;
     }
+
+    private void addDerivedResourceAttribute(Metacard metacard, ContentItem contentItem) {
+        Attribute attribute = metacard.getAttribute(Core.DERIVED_RESOURCE_URI);
+        if (attribute == null) {
+            attribute = new AttributeImpl(Core.DERIVED_RESOURCE_URI, contentItem.getUri());
+        } else {
+            AttributeImpl newAttribute = new AttributeImpl(attribute);
+            newAttribute.addValue(contentItem.getUri());
+            attribute = newAttribute;
+        }
+
+        metacard.setAttribute(attribute);
+    }
+
+    private int calculateOverviewHeight(BufferedImage image) {
+        final int width = image.getWidth();
+        final int height = image.getHeight();
+
+        if (width >= height) {
+            return (int) Math.round(height * (maxSideLength / width));
+        }
+
+        return Math.min(height, (int) maxSideLength);
+    }
+
+    private int calculateOverviewWidth(BufferedImage image) {
+        final int width = image.getWidth();
+        final int height = image.getHeight();
+
+        if (width >= height) {
+            return Math.min(width, (int) maxSideLength);
+        }
+
+        return (int) Math.round(width * (maxSideLength / height));
+    }
+
+    public void setMaxSideLength(int maxSideLength) {
+        if (maxSideLength > 0) {
+            LOGGER.debug("Setting derived image maxSideLength to {}", maxSideLength);
+            this.maxSideLength = maxSideLength;
+        } else {
+            LOGGER.warn(
```

Comment:
An `info` or `debug` level log is probably sufficient here. There's nothing for the admin to fix, necessarily, as everything will work just fine.